graphql/resolvers/User.js

```diff
@@ -1,19 +1,88 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+
+const isMeOrHasProfile = (user, me) => (
+  user._data.hasPublicProfile ||
+  Roles.userIsMe(user, me)
+)
+
 module.exports = {
-  initials (user) {
-    return user.name
-      .split(' ')
-      .map(p => p[0])
-      .join('')
-  },
-  publicUser (user, args, { req }) {
-    if (req.user.id === user.id) {
-      return user
+  email (user, args, { pgdb, user: me }) {
+    if (user._data.isEmailPublic || Roles.userIsMeOrInRoles(user, me, ['admin', 'supporter'])) {
+      return user._data.email
     }
     return null
   },
-  async address (user, args, {pgdb}) {
-    if (!user.addressId) return null
-    return pgdb.public.addresses.findOne({id: user.addressId})
+  async testimonial (user, args, { pgdb, user: me }) {
+    const testimonial = await pgdb.public.testimonials.findOne({
+      userId: user.id
+    })
+    if (
+      (testimonial.published && !testimonial.adminUnpublished) ||
+      (me && me.id === testimonial.userId)
+    ) {
+      return {
+        ...testimonial,
+        name: user.name
+      }
+    }
+  },
+  async latestComments (user, args, { pgdb, user: me }) {
+    if (isMeOrHasProfile(user, me)) {
+      return null
+    }
+    const userId = user.id
+    const limit = args.limit || 10
+
+    const comments = await pgdb.query(
+      `
+      SELECT
+        c.id,
+        c."userId",
+        c.content,
+        c."adminUnpublished",
+        c.published,
+        c."createdAt",
+        c."updatedAt",
+        c."discussionId",
+        d.title AS "discussionTitle"
+      FROM comments c
+      JOIN discussions d ON d.id = c."discussionId"
+      WHERE
+        c."userId" = :userId
+      ORDER BY
+        c."createdAt" DESC
+      LIMIT :limit;
+    `,
+      { userId, limit }
+    )
+
+    if (comments.length) {
+      return comments.map(comment => {
+        return {
+          ...comment,
+          discussion: {
+            id: comment.discussionId,
+            title: comment.discussionTitle
+          }
+        }
+      })
+    }
+  },
+  async credentials (user, args, { pgdb, user: me }) {
+    if (isMeOrHasProfile(user, me)) {
+      return pgdb.public.credentials.find({
+        userId: user.id
+      })
+    }
+  },
+  async address (user, args, {pgdb, user: me}) {
+    if (Roles.userIsMeOrInRoles(user, me, ['admin', 'supporter'])) {
+      if (!user._data.addressId) return null
```

Comment:
someone once told me to not use one-line if's, because of maintainability... who could that have been?